clang/test/CodeGenSYCL/remove-restriction-builtin-printf.cpp

```diff
@@ -10,7 +10,7 @@ int main() {
   q.submit([&](handler &h) {
     // CHECK: define {{.*}}spir_kernel {{.*}}
     h.single_task<class kernelA>([=]() {
-      // CHECK: @printf{{.*}}
+      // CHECK: call {{.*}}printf(ptr noundef nonnull dereferenceable(1) @{{.*}}, i32 noundef 24)
```

Comment:
I am okay with this change if it passes on Windows as well.   I think it does, but I can't tell.

Reply:
This will pass on windows i think:

LLVM IR for printf in Linux looks like : @printf(ptr noundef nonnull dereferenceable(1) @.str, i32 noundef 24) #2
and in Windows : @printf(ptr noundef nonnull dereferenceable(1) @"??_C@_0L@JKDODDCB@hello?0?5?$CFd?6?$AA@", i32 noundef 24) #2

Reply:
> This will pass on windows i think:

I think so too, but I have been bitten by accidental whitespace errors before.

Reply:
I have approved it. Please wait for approval from @premanandrao.